### PR TITLE
ref(ui): Improve buttons for dashboards edit

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Subtle changes visually, but the interactions should feel a lot better. Some small padding adjustments here and there 

Old
image

New
image

PS. I changed the thrash icon back to black, it is not red

### CODE REVIEW

File: static/app/views/dashboardsV2/dashboard.tsx [hunk 1]
```diff
@@ -577,9 +577,8 @@ class Dashboard extends Component<Props, State> {
           <ResizeHandle
```
Comment (evanpurkhiser) [inline static/app/views/dashboardsV2/dashboard.tsx:577]: Doesn't need to be a wrapper

File: static/app/views/dashboardsV2/dashboard.tsx [hunk 2]
```diff
@@ -686,8 +685,10 @@ const GridItem = styled('div')`
 const GridLayout = styled(WidthProvider(Responsive))`
   margin: -${space(2)};
 
-  .react-resizable-handle {
+  .react-grid-item > .react-resizable-handle {
     background-image: none;
+    width: auto;
```
Comment (evanpurkhiser) [inline static/app/views/dashboardsV2/dashboard.tsx:690]: Allow the size of the icon to set the size

File: static/app/views/dashboardsV2/dashboard.tsx [hunk 3]
```diff
@@ -696,12 +697,14 @@ const GridLayout = styled(WidthProvider(Responsive))`
 
   .react-grid-item.react-grid-placeholder {
     background: ${p => p.theme.purple200};
+    border-radius: ${p => p.theme.borderRadius};
```
Comment (evanpurkhiser) [inline static/app/views/dashboardsV2/dashboard.tsx:700]: Add a border-radius to the grid placeholder when dragging it around.

File: static/app/views/dashboardsV2/widgetCard/index.tsx [hunk 4]
```diff
@@ -74,24 +75,39 @@ class WidgetCard extends React.Component<Props> {
...
+              size="xsmall"
+              aria-label={t('Drag Widget')}
+              icon={<IconGrabbable />}
+              borderless
+              className={DRAG_HANDLE_CLASS}
+              {...draggableProps?.listeners}
+              {...draggableProps?.attributes}
+            />
           )}
-          <IconClick data-test-id="widget-edit" onClick={onEdit}>
```
Comment (evanpurkhiser) [inline static/app/views/dashboardsV2/widgetCard/index.tsx:86]: Get rid of onClicks on not buttons

File: static/app/views/dashboardsV2/dashboard.tsx [hunk 5]
```diff
@@ -568,18 +570,21 @@ class Dashboard extends Component<Props, State> {
         rowHeight={ROW_HEIGHT}
         margin={WIDGET_MARGINS}
         draggableHandle={`.${DRAG_HANDLE_CLASS}`}
+        draggableCancel={`.${DRAG_RESIZE_CLASS}`}
```
Comment (evanpurkhiser) [inline static/app/views/dashboardsV2/dashboard.tsx:573]: This is a bit better than trying to override all the styles on `react-resizable-handle`

File: static/app/views/dashboardsV2/widgetCard/index.tsx [hunk 6]
```diff
@@ -264,6 +270,7 @@ const WidgetTitle = styled(HeaderTitle)`
 
 const WidgetHeader = styled('div')`
   padding: ${space(2)} ${space(1)} 0 ${space(3)};
+  min-height: 36px;
```
Comment (evanpurkhiser) [inline static/app/views/dashboardsV2/widgetCard/index.tsx:273]: This fixes a small height change when going into edit mode.

File: static/app/views/dashboardsV2/dashboard.tsx [hunk 7]
```diff
@@ -426,15 +428,15 @@ class Dashboard extends Component<Props, State> {
       const key = constructGridItemKey(widget);
       const dragId = key;
       return (
-        <GridItem key={key} data-grid={widget.layout}>
+        <div key={key} data-grid={widget.layout}>
```
Comment (priscilawebdev) [inline static/app/views/dashboardsV2/dashboard.tsx:431]: is `data-grid` a valid HTML attribute?

File: static/app/views/dashboardsV2/dashboard.tsx [hunk 8]
```diff
@@ -568,18 +570,21 @@ class Dashboard extends Component<Props, State> {
...
+        draggableCancel={`.${DRAG_RESIZE_CLASS}`}
         layouts={layouts}
         onLayoutChange={this.handleLayoutChange}
         onBreakpointChange={this.handleBreakpointChange}
         isDraggable={canModifyLayout}
         isResizable={canModifyLayout}
         resizeHandle={
           <ResizeHandle
-            className="react-resizable-handle"
+            aria-label="Resize Widget"
```
Comment (priscilawebdev) [inline static/app/views/dashboardsV2/dashboard.tsx:581]:
```suggestion
            aria-label={t("Resize Widget")}
```

File: static/app/views/dashboardsV2/dashboard.tsx (hunk 7 above)
Comment (evanpurkhiser) [reply static/app/views/dashboardsV2/dashboard.tsx:431]: yep, any `data-*` attribute is valid!

**Comment by benvinegar** [comment]
Red seems like a little much? Do we do that elsewhere?

**Comment by evanpurkhiser** [comment]
> Red seems like a little much? Do we do that elsewhere?

Cc @vuluongj20 @doralchan

File: static/app/views/dashboardsV2/dashboard.tsx (hunk 7 above)
Comment (priscilawebdev) [reply static/app/views/dashboardsV2/dashboard.tsx:434]: TIL

**Comment by evanpurkhiser** [comment]
OK killed off the red in this

**Review by narsaynorath — APPROVED** [review]
Looks good to me. I like that it was able to get rid of some of the custom styling for the `react-resizable-handle` class, and it's more consistent with the other buttons in the header. Thanks for putting up this change :+1:

**Review by edwardgou-sentry — APPROVED** [review]

**Pull request merged**